### ISSUE TITLE
Sensor can't be showed on cloud portal with iot-rest-api-server@0.6.0

### ISSUE THREAD

**Issue by xiaoyu3x** [01org/SmartHome-Demo#185]
## Description:

Sensor can't be showed on cloud portal when using iot-rest-api-server@0.6.0.

## Steps:
1.Install iot-rest-api-server to 0.6.0 and launch
```
git clone https://github.com/01org/iot-rest-api-server.git 
cd iot-rest-api-server
npm install
cd bin
./iot-rest-api-server.js
```
2.Build the  [led.js](https://github.com/01org/SmartHome-Demo/blob/master/ocf-servers/zjs-servers/led.js) and and flash it to set up the 6lowpan.
3.Try to discover the OCF resources: `cd iot-rest-api-server/test && ./oic-get /res?rt=oic.r.switch.binary `
4.Build the Docker container and run docker cloud.

## Actual Result:
Led sensor can't be showed on cloud portal.

## Expect Result:
Sensor can be showed on cloud portal

**Comment by wanghongjuan** [comment]
@ichbinblau @ttzeng @gvancuts Could you please take a look this issue?

**Comment by gvancuts** [comment]
Hi @wanghongjuan , could you send us a log from the container (see paragraph 8 in [this document](https://github.com/intel/SmartHome-Demo/blob/master/smarthome-web-portal/tools/docker/README.md) for instructions)?

**Comment by xiaoyu3x** [comment]
hello, @gvancuts 
Sorry,  I upload container log late.
The following is container log and I flash led.js into A101.
Iot-rest-api-server can get led resource by command `./oic-get /res?rt=oic.r.switch.binary`, but web-portal can't get led resource and it can't get led resource by this way.
[docker.log](https://github.com/intel/SmartHome-Demo/files/1672086/docker.log)

**Comment by ichbinblau** [comment]
Hi, @xiaoyu3x  Have you set the IoT rest service url in the admin portal? Telling from the log, celery worker failed to reach the rest service. 

> [2018-01-29 03:05:08,338: INFO/ForkPoolWorker-3] http://172.17.0.1:8000/api/oic/res
[2018-01-29 03:05:08,339: INFO/ForkPoolWorker-3] Starting new HTTP connection (1): 172.17.0.1
[2018-01-29 03:05:08,340: ERROR/ForkPoolWorker-3] <class 'requests.exceptions.ConnectionError'> HTTPConnectionPool(host='172.17.0.1', port=8000): Max retries exceeded with url: /api/oic/res (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f8da01fadd0>: Failed to establish a new connection: [Errno 111] Connection refused',))

**Comment by xiaoyu3x** [comment]
Sorry, @ichbinblau 
Just log content is not clear. I get a new log when web portal runs normally. 
The following is container log
[docker.log](https://github.com/intel/SmartHome-Demo/files/1672467/docker.log)

You can see docker retrieve resource normally and /a/led can be found.


```
[2018-01-29 05:34:54,074: INFO/ForkPoolWorker-2] Starting new HTTP connection (1): 172.17.0.1
[2018-01-29 05:34:59,077: INFO/ForkPoolWorker-2] CeleryTask.tasks.get_sensor_data[2287ff36-1703-491c-ba64-8414891eb982]: Update device list: {'680e7a44-86c4-f96a-b039-7188b2588b6c': 'IoT REST API Server', '8bca6209-dbca-4209-6bcb-62097bcb6209': 'ZJS Device'}
[2018-01-29 05:34:59,171: INFO/ForkPoolWorker-2] CeleryTask.tasks.get_sensor_data[2287ff36-1703-491c-ba64-8414891eb982]: Resource {'status': True, 'gateway_id': 1, 'uuid': u'8bca6209-dbca-4209-6bcb-62097bcb6209', 'observable': False, 'created_at': '2018-01-29 05:34:59', 'tag': 'ZJS Device', 'sensor_type_id': 16, 'path': '/oc/wk/introspection', 'id': 1} is added.
```
But I can't see LED sensor on web portal.
![selection_011](https://user-images.githubusercontent.com/17694094/35496721-367a67c4-0502-11e8-9dd6-06427f70d64f.png)
![selection_012](https://user-images.githubusercontent.com/17694094/35496730-3c08ee54-0502-11e8-9e5c-d33bb1fa360c.png)
![selection_013](https://user-images.githubusercontent.com/17694094/35496736-40364738-0502-11e8-940c-b7e39c8a900c.png)

**Comment by ichbinblau** [comment]
@xiaoyu3x I found the path (/oc/wk/introspection) is not an known resource path. Would you also send me the output of the /oic/api/res? 

> [2018-01-29 05:34:59,171: INFO/ForkPoolWorker-2] CeleryTask.tasks.get_sensor_data[2287ff36-1703-491c-ba64-8414891eb982]: Resource {'status': True, 'gateway_id': 1, 'uuid': u'8bca6209-dbca-4209-6bcb-62097bcb6209', 'observable': False, 'created_at': '2018-01-29 05:34:59', 'tag': 'ZJS Device', 'sensor_type_id': 16, 'path': '/oc/wk/introspection', 'id': 1} is added.

**Comment by xiaoyu3x** [comment]
hi, @ichbinblau 
The following is localhost:8000/api/oic/res result.
![selection_015](https://user-images.githubusercontent.com/17694094/35499341-e2c3e99c-050c-11e8-86d2-8ab40573cade.png)
![selection_016](https://user-images.githubusercontent.com/17694094/35499354-ebc0bcdc-050c-11e8-85e2-ccb6d656c2e1.png)

Why there are two kinds of results? This issue's key is iot-rest-api-server hasn't support `./oic-get /res` and `localhost:8000/api/oic/res` method by [comment ](https://github.com/intel/zephyr.js/issues/1548#issuecomment-331147561)
It is easy to cause the discovery failure when using above two way to retrieve resource.
So we use `./oic-get /res?rt=**` to retrieve specific resource.
I think web portal retrieves resource by './oic-get /res', so it cause iot-rest-api-server break off.

**Comment by uhofemei** [comment]
@xiaoyu3x which version of Zephyr are you testing this with? I received an update from one of our ZJS developers stating that Bluetooth in Zephyr version 1.10 should be more stable. Would you have cycles to test this? intel/zephyr.js#1237 (comment)

**Comment by jimmy-huang** [comment]
@xiaoyu3x is probably right, ZJS can longer support get /res to retrieve all resources at the same time, because the string that encapsulates all the metadata is not going to work on the A101 due to memory constraints, so the best way is to change the web portal to retrieve the resources individually using get /res?rt= to ensure that you get a response.

**Comment by uhofemei** [comment]
This would mean the celery/flask queue engine code that retrieves the sensor data would have to be re-written to fetch the sensor data. @gvancuts, what do you think?

**Comment by gvancuts** [comment]
We _could_ change our code to accomodate this limitation. *But* it does make me wonder if that behaviour is appropriate. Shouldn't this be fixed in ZJS instead?

I was also trying (not very hard) to follow the thread that @xiaoyu3x pointed at earlier (see https://github.com/intel/zephyr.js/issues/1548#issuecomment-331147561) and I thought that one had been fixed. If so, are we sure we have identified the problem correctly?

**Comment by jimmy-huang** [comment]
@gvancuts This is just a hardware limitation with running A101 given the limited num of RAM.  There's nothing we could do on ZJS, so I think the options are limited if you are planning to use the A101 to work around these issues.  However, I do agree that we first need to identify if the problem to make sure this is the appropriate fix.

**Comment by jimmy-huang** [comment]
on a side note, have you guys tried increasing ram settings by setting RAM=70 or something higher? I think it defaults to 64k, so 16k is left for ARC, but I am sure the ARC uses less than that, so you can pull off a few K ram into the X86 side.  At least if you can get enough to store the the JSON string that's being returned to the client.

**Comment by gvancuts** [comment]
@jimmy-huang , is there any kind of error we could return to indicate that the answer won't be valid?

**Comment by jimmy-huang** [comment]
@gvancuts  I don't remember the details of the bug but I believe it was segfaulting, so i don't know if had a chance to actually send anything back to the client, it appeared looking at the log that the server just died, and the iot-rest-server got an empty list of elements back.

**Comment by xiaoyu3x** [comment]
hello, @jimmy-huang 

Is there new progress about this issue？

**Comment by jimmy-huang** [comment]
@xiaoyu3x no, there's not much progress in OCF and Zephyr, did you try the updated RAM settings I mentioned? like building with RAM=74 or something?

**Comment by xiaoyu3x** [comment]
hello, @jimmy-huang 
Thanks for your advice. 
According to your suggestion, I set RAM=74 and more.
I obtained the following result on iot-rest-api-server when RAM=74:
![selection_001](https://user-images.githubusercontent.com/17694094/37071576-19f3e4dc-21f8-11e8-8e68-570995011ef1.png)
I couldn't use command `oic-get /res` to retrieve sensors that I want to select resources.
I only retrieve expect resource by 'resourcetype' method.

To resolve above phenomenon, I set RAM=75 ...
but code can't work normally, I think ARC is too small.
![selection_002](https://user-images.githubusercontent.com/17694094/37071795-4815a412-21f9-11e8-8543-3ad772c7fdb1.png)

**Comment by jimmy-huang** [comment]
Yeah, if you are trying to use get /res without specifying a type, then RAM is really important, and there's nothing you can really do if you use the a101.

**Comment by xiaoyu3x** [comment]
hello, @jimmy-huang 
I just used `/res` and ` /res?rt=` separately to test FRDM K64F board.
Only Flash led sensor into k64f, I find led resource can be retrieve successfully in two ways.
![selection_003](https://user-images.githubusercontent.com/17694094/37137021-f8a0135e-22de-11e8-8801-4efe0f732470.png)

Flash multi sensors into k64f, can find resource by ` /res?rt=` way, `/res ` way can't retrieve any sensor.
![selection_004](https://user-images.githubusercontent.com/17694094/37137134-6d343c7c-22df-11e8-911a-ffbb9a315a31.png)
